src/server/services/user.service.ts

```diff
@@ -732,26 +732,44 @@ export const createUserReferral = async ({
 }) => {
   const user = await dbRead.user.findUniqueOrThrow({
     where: { id },
-    select: { id: true, referral: { select: { id: true } } },
+    select: { id: true, referral: { select: { id: true, userReferralCodeId: true } } },
   });
 
-  if (!!user.referral) {
+  if (!!user.referral?.userReferralCodeId || (!!user.referral && !userReferralCode)) {
     return;
   }
 
   if (userReferralCode || source) {
     // Confirm userReferralCode is valid:
-    const referralCode = await dbRead.userReferralCode.findFirst({
-      where: { code: userReferralCode },
-    });
-
-    // Create new referral:
-    await dbWrite.userReferral.create({
-      data: {
-        userId: id,
-        source,
-        userReferralCodeId: referralCode?.id ?? undefined,
-      },
-    });
+    const referralCode = !!userReferralCode
+      ? await dbRead.userReferralCode.findFirst({
+          where: { code: userReferralCode },
+        })
```

Comment:
If we are using `findFirst`, then we should add the condition to check if it's not deleted I think
```suggestion
      ? await dbRead.userReferralCode.findFirst({
          where: { code: userReferralCode, deletedAt: null },
        })
```